File: benchmark/datastruct/Datastruct.java
class Point {
    long x, y, z;
    Point(long x, long y, long z) {
        this.x = x; this.y = y; this.z = z;
    }
}

public class Datastruct {
    public static void main(String[] args) {
        int iterations = 10_000_000;
        long totalSum = 0;

        long start = System.nanoTime();

        for (int i = 0; i < iterations; i++) {
            // New heap allocation + Object Header + Pointer reference
            Point p = new Point(i, 1, 1);
            totalSum += p.x + p.y + p.z;
        }

        long end = System.nanoTime();
        System.out.println("Java (Heap Object): " + (end - start) / 1e9 + "s");
        System.out.println("Final Sum: " + totalSum);
    }
}

File: benchmark/datastruct/datastruct.py
import time

class Point:
    def __init__(self, x, y, z):
        self.x = x
        self.y = y
        self.z = z

def run_benchmark():
    iterations = 10_000_000
    total_sum = 0
    
    start = time.perf_counter()
    
    for i in range(iterations):
        # Heavy heap allocation and dynamic attribute binding
        p = Point(i, 1, 1)
        total_sum += p.x + p.y + p.z
        
    end = time.perf_counter()
    
    print(f"Python (Legacy Class): {end - start:.4f}s")
    print(f"Final Sum: {total_sum}")

if __name__ == "__main__":
    run_benchmark()